### PR TITLE
[release/5.0] Move to OSX.1015.Amd64.Open

### PR DESCRIPTION
**Pull request by dougbu**
nits:
- remove version from macOS job names
- remove mention of of macOS 10.14 in RunHelix.ps1

### CODE REVIEW

File: eng/targets/Helix.Common.props [hunk 1]
```diff
@@ -22,7 +22,7 @@
   <ItemGroup Condition="'$(IsRequiredCheck)' == 'true' AND '$(TargetArchitecture)' == 'x64'">
     <HelixAvailableTargetQueue Include="Ubuntu.1804.Amd64.Open" Platform="Linux" />
     <HelixAvailableTargetQueue Include="Windows.10.Amd64.Open" Platform="Windows" />
-    <HelixAvailableTargetQueue Include="OSX.1014.Amd64.Open" Platform="OSX" />
+    <HelixAvailableTargetQueue Include="OSX.1100.Amd64.Open" Platform="OSX" />
```
Comment (dougbu) [inline eng/targets/Helix.Common.props:25]: In release/6.0 and main, we test on 10.15 in the full matrix (scheduled) runs. Does not seem necessary here. Any objections @dotnet/aspnet-build❔

Comment (wtgodbe) [reply eng/targets/Helix.Common.props:25]: Fine by me

**Review by wtgodbe — APPROVED** [review]

**Comment by dougbu** [comment]
Switched to use macOS 10.15 Helix queue rather than diving deeper into problems seen w/ the Razor integration tests on macOS 11

**Pull request merged**